bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java

```diff
@@ -0,0 +1,376 @@
+
+/**
+ * The {@link SnmpTargetHandler} is responsible for handling commands, which are
+ * sent to one of the channels or update remote channels
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class SnmpTargetHandler extends BaseThingHandler implements ResponseListener, CommandResponder {
+    private final Logger logger = LoggerFactory.getLogger(SnmpTargetHandler.class);
+
+    private @NonNullByDefault({}) SnmpTargetConfiguration config;
+    private final SnmpService snmpService;
+    private @NonNullByDefault({}) ScheduledFuture<?> refresh;
+    private int timeoutCounter = 0;
+
+    private @NonNullByDefault({}) AbstractTarget target;
+    private @NonNullByDefault({}) String targetAddressString;
+
+    private @NonNullByDefault({}) Set<SnmpChannelConfiguration> readChannelSet;
+    private @NonNullByDefault({}) Set<SnmpChannelConfiguration> writeChannelSet;
+    private @NonNullByDefault({}) Set<SnmpChannelConfiguration> trapChannelSet;
+
+    public SnmpTargetHandler(Thing thing, SnmpService snmpService) {
+        super(thing);
+        this.snmpService = snmpService;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                SnmpChannelConfiguration channel = readChannelSet.stream().filter(c -> channelUID.equals(c._channelUID))
+                        .findFirst().orElseThrow(() -> new IllegalArgumentException("no writable channel found"));
+                PDU pdu = new PDU(PDU.GET, Collections.singletonList(new VariableBinding(channel._oid)));
+                snmpService.send(pdu, target, null, this);
+            } else if (command instanceof DecimalType || command instanceof StringType
+                    || command instanceof OnOffType) {
+                SnmpChannelConfiguration channel = writeChannelSet.stream()
+                        .filter(config -> channelUID.equals(config._channelUID)).findFirst()
+                        .orElseThrow(() -> new IllegalArgumentException("no writable channel found"));
+                Variable variable;
+                if (command instanceof OnOffType) {
+                    variable = OnOffType.ON.equals(command) ? channel._onValue : channel._offValue;
+                    if (variable == null) {
+                        logger.debug("skipping {} to {}: no value defined", command, channelUID);
+                        return;
+                    }
+                } else {
+                    variable = convertDatatype(command, channel.datatype);
+                }
+                PDU pdu = new PDU(PDU.SET, Collections.singletonList(new VariableBinding(channel._oid, variable)));
+                snmpService.send(pdu, target, null, this);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.warn("can't process command {} to {}: {}", command, channelUID, e.getMessage());
```

Comment:
Should you just log e?

Reply:
Why would that be better? We get a good reason in the message, the stacktrace is not helpful and just fills the logs.

Reply:
You can’t guarantee that it’s not interesting  because it’s a standard exception from the JVM. But I do agree that it’s most likely not a relevant trace.